frontend/src/main-components/CreateCardContent.js

```diff
@@ -15,27 +15,60 @@ class CreateCardContent extends Component {
       fromLang: "none",
       toLang: "none",
       folder: "",
+      uploadUrlFetched: false,
+      imageUploadUrl: "",
     };
     this.translateText = this.translateText.bind(this);
     this.fromLangSelected = this.fromLangSelected.bind(this);
     this.toLangSelected = this.toLangSelected.bind(this);
+    this.imageSelected = this.imageSelected.bind(this);
+    this.imageUploadUrl = "";
+  }
+
+  async componentDidMount() {
+    try {
+      const uploadUrl = await fetch("/upload").then((response) =>
+        response.text()
+      );
+      if (!uploadUrl.ok) {
+        throw Error(uploadUrl.statusText);
+      }
+      this.setState({ imageUploadUrl: uploadUrl, uploadUrlFetched: true });
+    } catch (error) {
+      alert("Refresh page to create a card");
+    }
   }
 
   /**
    * When the user has finished typing, the Google Translate
    * API is called to fetch the translated version of the text input.
    */
   translateText(event) {
+    const text = event.target.value;
     // Ensures that languages have been selected before translating
     if (this.state.fromLang !== "none" && this.state.toLang !== "none") {
-      const translated = getTranslation(
-        event.target.value,
-        this.state.fromLang,
-        this.state.toLang
-      );
-      this.setState({ translation: translated, text: event.target.value });
+      (async () => {
+        try {
+          const translated = await getTranslation(
+            text,
+            this.state.fromLang,
+            this.state.toLang
+          );
+          // Checks if the getTranslation function encountered an error
+          if (!translated.ok) {
+            throw Error(translated.statusText);
+          }
+          this.setState({ translation: translated, text: text });
+        } catch (error) {
+          this.setState({
+            translation: "Text could not be translated",
+            text: text,
+          });
+        }
+      })();
+    } else {
+      this.setState({ text: text });
```

Comment:
Small nit: when you have a variable with the same name as the key in the object, there's a shortcut:

```
      this.setState({ text });
```

Reply:
This is a cool tip, thank you.

Reply:
done